### PR TITLE
[stable25] Add required accessibility attributes to legacy files navigation

### LINKED ISSUE #37101: [BITV] 9.4.1.2/3.4 - The "File settings" button is an accordion button that should be implemented as such with the aria-expanded attribute. (2)
See https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/ for further details. 

- [x] Fix
- [x] Backport

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v3-n4

</details>

- [ ] 9.4.1.2/10.4 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v10-n4

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: #37101
* Already fixed for NC26 and NC27 with 887c9e05de88f81ed6f0cb88bd185c05b1a22076

## Summary
The files settings button on the app navigation is an accordion button for NC25, so it requires some additional accessibility attributes to be set.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

**Comment by Pytal** [comment]
Thanks @susnux!

I believe we don't need to do this for 25, can you confirm @AndyScherzinger?

**Comment by AndyScherzinger** [comment]
Can confirm, having it for the versions above as described by @susnux would be enough

**Review by JuliaKirschenheuter — APPROVED** [review]

**Comment by susnux** [comment]
> I believe we don't need to do this for 25

Are there any objections to fix this for NC25 / merge? (As the work is already done 😅 )

**Comment by AndyScherzinger** [comment]
@susnux if the work is done _and_ tested to work _and_ isn't breaking the navigation for apps I'd be fine.

Any objections or comments @skjnldsv ?

**Review by skjnldsv — APPROVED** [review]

File: apps/files/src/files-app-settings.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ window.addEventListener('DOMContentLoaded', function() {
 		appSettingsHeader.addEventListener('click', e => {
 			const opened = e.currentTarget.children[0].classList.contains('opened')
 			OCA.Files.Settings.settings.forEach(e => opened ? e.close() : e.open())
+			e.currentTarget.children[0].setAttribute('aria-expanded', opened ? 'false' : 'true')
```
Comment (skjnldsv) [inline apps/files/src/files-app-settings.js:55]: > Are there any objections to fix this for NC25 / merge? (As the work is already done sweat_smile )

I would suggest to be a little more careful on this line and make sure it cannot fail

```suggestion
			e.currentTarget?.children?.[0]?.setAttribute?.('aria-expanded', opened ? 'false' : 'true')
```

**Pull request merged**